RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -548,14 +546,46 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
     } else if (nnlsWork_.bxs.coeff(iBX) == pedestalBX_) {
       mdi.pedEnergy = nnlsWork_.ampVec.coeff(iBX);
     } else if (nnlsWork_.bxs.coeff(iBX) == -1) {
-      mdi.pEnergy = nnlsWork_.ampVec.coeff(iBX);
+      mdi.p1Energy = 0;
+      mdi.p1Energy = nnlsWork_.ampVec.coeff(iBX);
       for (unsigned int iTS = 0; iTS < nnlsWork_.tsSize; ++iTS) {
-        mdi.pPulse[iTS] = nnlsWork_.pulseMat.col(iBX).coeff(iTS);
+        mdi.p1Pulse[iTS] = nnlsWork_.pulseMat.col(iBX).coeff(iTS);
       }
     } else if (nnlsWork_.bxs.coeff(iBX) == 1) {
-      mdi.nEnergy = nnlsWork_.ampVec.coeff(iBX);
+      mdi.n1Energy = 0;
+      mdi.n1Energy = nnlsWork_.ampVec.coeff(iBX);
+      for (unsigned int iTS = 0; iTS < nnlsWork_.tsSize; ++iTS) {
+        mdi.n1Pulse[iTS] = nnlsWork_.pulseMat.col(iBX).coeff(iTS);
+      }
+    } else if (nnlsWork_.bxs.coeff(iBX) == -2) {
+      mdi.p2Energy = 0;
```

Comment:
this pattern seems to be a good case to replace n[1-4]\* and p[1-3]\* with an array of size 7 so that all of the code can fit in 5-7 lines instead of 40

Reply:
Replaced n[1-4]* and p[1-3]* with **ootEnergy[7]** and **ootPulse[7][MaxSVSize]**.